packages/client/modules/teamDashboard/components/TeamArchive/TeamArchive.tsx

```diff
@@ -278,79 +287,108 @@ const TeamArchive = (props: Props) => {
-          {edges.length ? (
-            <CardGrid>
-              <InfiniteLoader
-                isRowLoaded={isRowLoaded}
-                loadMoreRows={maybeLoadMore}
-                rowCount={MathEnum.MAX_INT}
-              >
-                {({onRowsRendered, registerChild}) => {
-                  _onRowsRenderedRef.current = onRowsRendered
-                  return (
-                    <div style={{flex: '1 1 auto'}}>
-                      <AutoSizer>
-                        {({height, width}) => {
-                          return (
-                            <Grid
-                              cellRenderer={rowRenderer}
-                              columnCount={columnCount}
-                              columnWidth={CARD_WIDTH}
-                              deferredMeasurementCache={cellCache}
-                              estimatedColumnSize={CARD_WIDTH}
-                              estimatedRowSize={182}
-                              height={height}
-                              onRowsRendered={onRowsRendered}
-                              onSectionRendered={_onSectionRendered}
-                              ref={(c) => {
-                                gridRef.current = c
-                                registerChild(c)
+    <ClassNames>
+      {({css}) => {
+        const noMoreMessage = css({
+          '::after': {
+            content: `"🎉 That's all folks! There are no further tasks in the archive."`,
+            backgroundColor: '#FFFFFF',
+            border: `1px solid ${PALETTE.SLATE_400}`,
+            borderRadius: 4,
+            fontSize: 14,
+            display: 'inline-block',
+            padding: 16,
+            position: 'absolute',
+            left: Math.floor((columnCount * CARD_WIDTH) / 2) + 'px',
+            transform: 'translateX(-50%)'
+          }
+        })
+        return (
+          <>
+            {!returnToTeamId && <UserTasksHeader viewerRef={viewer} />}
+            <Root>
+              {returnToTeamId && (
+                <Header>
+                  <TeamArchiveHeader teamId={returnToTeamId} />
+                  <Border />
+                </Header>
+              )}
+              <Body>
+                {edges.length ? (
+                  <CardGrid>
+                    <InfiniteLoader
+                      isRowLoaded={isRowLoaded}
+                      loadMoreRows={maybeLoadMore}
+                      rowCount={MathEnum.MAX_INT}
+                    >
+                      {({onRowsRendered, registerChild}) => {
+                        _onRowsRenderedRef.current = onRowsRendered
+                        return (
+                          <div style={{flex: '1 1 auto'}}>
+                            <AutoSizer>
+                              {({height, width}) => {
+                                return (
+                                  <Grid
+                                    className={!hasNext && isScrolledToBottom ? noMoreMessage : ''}
```

Comment:
I believe this is redundant checking.
Because there are indicators of a different kinds (server and client). `hasNext` comes from the server's response (connection data), which indicates that there is no more element from the server's point of view.
`isLoadingNext` refers to the status of a request on the client (that we are waiting for the answer to the next piece of data).

After the server responds that there are no more items to load, the client cannot make a pagination request for the next items.
So when `hasNext` is false, `isLoadingNext` is always false too.